ComponentKit/Core/Scope/CKComponentScopeFrame.mm

```diff
@@ -88,4 +88,29 @@ - (instancetype)initWithHandle:(CKComponentScopeHandle *)handle
   return self;
 }
 
+- (CKComponentScopeFrame *)scopeFrameForClass:(Class)componentClass identifier:(id)identifier;
+{
+  auto it = _children.find({componentClass, identifier});
```

Comment:
This is going to be relatively less efficient when you're doing it based on global identifiers instead of class/identifiers; now it'll be an O(N) search instead of O(1). But given that the number of children will typically be ~5 at most, that should be just fine.